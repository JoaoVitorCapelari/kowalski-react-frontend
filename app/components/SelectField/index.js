/**
*
* SelectField
* options prop: An object containing:
* 'labelField'
*   - the field in the elements of items that should be fetched in order to show the text of the option. Defaults to label,
* 'valueField'
*   - the field in the elements of items that should be fetched in order to show the value of the option. Defaults to value,
*  'items'
*   - The array of objects representing the option elements of the select input
*/

import React from 'react';
import PropTypes from 'prop-types';

function SelectField({ input, label, id, children, meta: { touched, error } }) {
  const labelHtml = label ? <label htmlFor={input.id} className="label">{label}</label> : '';
  const errorSelectClasses = touched && error ? 'is-danger' : '';
  return (
    <div className="field">
      {labelHtml}
      <span className="control">
        <div className={`select ${errorSelectClasses}`}>
          <select {...input} id={id}>
            { children }
          </select>
        </div>
        {touched && error && <p className="help is-danger">{error}</p>}
      </span>
    </div>
  );
}

SelectField.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
  children: PropTypes.any,
  id: PropTypes.string,
};

export default SelectField;
