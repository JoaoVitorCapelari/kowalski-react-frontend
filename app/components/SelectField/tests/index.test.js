import React from 'react';
import { shallow } from 'enzyme';

import SelectField from '../index';

describe('<SelectField />', () => {
  it('renders', () => {
    const renderedComponent = shallow(
      <SelectField meta={{ touched: false }} input={{}} />
    );
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
