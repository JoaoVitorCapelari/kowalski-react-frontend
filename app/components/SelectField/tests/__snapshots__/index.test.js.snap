// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectField /> renders 1`] = `"<div class=\\"field\\"><span class=\\"control\\"><div class=\\"select \\"><select></select></div></span></div>"`;
